raml-parser-interface/src/main/java/org/mule/apikit/common/ValidationUtils.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.apikit.common;
+
+import org.mule.apikit.model.parameter.Parameter;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static java.lang.String.valueOf;
+
+public class ValidationUtils {
+
+  public static boolean validateQueryParams(Map<String, List<String>> queryParams, Map<String, Parameter> facets,
+                                            List<QueryStringGroup> queryStringGroups) {
+    List<String> validatedQueryParamsKeys = new ArrayList<>();
+    for (Map.Entry<String, List<String>> queryParam : queryParams.entrySet()) {
+      String value;
+      Parameter expected = facets.get(queryParam.getKey());
+      if (expected != null) {
+        List<String> values = queryParam.getValue();
+        if (values.size() > 1 || expected.isArray()) {
+          value = toArrayYamlRepresentation(queryParam, expected);
+        } else {
+          value = values.isEmpty() ? null : values.get(0);
+        }
+
+        if (!expected.validate(value)) {
+          return false;
+        }
+      }
+      validatedQueryParamsKeys.add(queryParam.getKey());
+    }
+    for (QueryStringGroup group : queryStringGroups) {
+      if (validatedQueryParamsKeys.containsAll(group.getRequiredProperties())
+          && (group.supportAdditionalProperties() || !group.hasAdditionalProperties(validatedQueryParamsKeys))) {
```

Comment:
group.supportAdditionalProperties() can be checked only when the sizes are different

Reply:
size can be equal and have different values since the check is only for required